plugins/nordic-hid/fu-nordic-hid-cfg-channel.c

```diff
@@ -811,49 +909,49 @@ fu_nordic_hid_cfg_channel_dfu_start(FuNordicDeviceCfgChannel *self,
 static gboolean
 fu_nordic_hid_cfg_channel_probe(FuDevice *device, GError **error)
 {
+	FuNordicHidCfgChannel *self = FU_NORDIC_HID_CFG_CHANNEL(device);
 	/* FuUdevDevice->probe */
 	if (!FU_DEVICE_CLASS(fu_nordic_hid_cfg_channel_parent_class)->probe(device, error))
 		return FALSE;
 
-	return fu_udev_device_set_physical_id(FU_UDEV_DEVICE(device), "hid", error);
+	if (self->peer_id == 0)
+		self->udev = FU_UDEV_DEVICE(self);
+	else {
+		FuDevice *parent = fu_device_get_parent(device);
+		self->udev = FU_UDEV_DEVICE(parent);
+	}
```

Comment:
hmm, this is a bit murky with regard to lifetimes; I'd much rather have a private flag set here for the "use parent" case and then a "getter" for when we need the raw udev_device -- i.e. not store it in the structure either.

Reply:
hmmm.... orks well with the current structure since the object is declared with `G_DECLARE_FINAL_TYPE`. Tested with several devices connected, including 2 via dongles.
The only caveat for me was the flag `FU_DEVICE_INTERNAL_FLAG_USE_PARENT_FOR_OPEN` needed to avoid udev_device closing prior the flashing

Reply:
I've pushed 3 fixup commits -- if you can rebase against master and squash them in to the right places this LGTM, thanks.

Reply:
Thanks! done.